pkg/inject/report.go

```diff
@@ -47,10 +54,10 @@ func (r Report) Injectable() bool {
 
 // update updates the report for the provided resource conf.
 func (r *Report) update(conf *ResourceConfig) {
-	r.InjectDisabled = conf.podMeta.ObjectMeta.GetAnnotations()[k8s.ProxyInjectAnnotation] == k8s.ProxyInjectDisabled
-	r.HostNetwork = conf.podSpec.HostNetwork
-	r.Sidecar = healthcheck.HasExistingSidecars(conf.podSpec)
-	r.UDP = checkUDPPorts(conf.podSpec)
+	r.InjectDisabled = conf.pod.meta.ObjectMeta.GetAnnotations()[k8s.ProxyInjectAnnotation] == k8s.ProxyInjectDisabled
+	r.HostNetwork = conf.pod.spec.HostNetwork
+	r.Sidecar = healthcheck.HasExistingSidecars(conf.pod.spec)
+	r.UDP = checkUDPPorts(conf.pod.spec)
 }
```

Comment:
This func now duplicates a bunch of code that exists in pkg/inject/inject.go. Can you instead write it like:

```
func (r *Report) update(conf *ResourceConfig) {
	r.InjectDisabled = conf.InjectDisabled()
	r.HostNetwork = conf.PodUsingHostNetwork()
	r.Sidecar = conf.HasExistingProxy()
	r.UDP = checkUDPPorts(conf.pod.spec)
}
```